tests/testcontrol.py

```diff
@@ -138,15 +37,116 @@
 class TestCourierControl(unittest.TestCase):
 
     def setUp(self):
-        os.mkdir('tmp')
-        os.system('cp -a queuefiles tmp/queuefiles')
-        os.system('cp -a configfiles tmp/configfiles')
+        self.tmpdir = tempfile.mkdtemp()
+        shutil.copytree(f"{os.path.dirname(__file__)}/queuefiles", f"{self.tmpdir}/queuefiles")
+        shutil.copytree(f"{os.path.dirname(__file__)}/configfiles", f"{self.tmpdir}/configfiles")
+        courier.config.sysconfdir = f'{self.tmpdir}/configfiles'
+
+        self.message = {}
+        self.message['xalias'] = {'control_paths': [self.tmpdir + '/queuefiles/control-xalias'],
```

Comment:
Would it be possible to leave the bulk of that dict in place and simply update control_paths in setUp?

Reply:
Probably. I thought originally about that, but it seemed uglier to me.
One would have to add a placeholder to the path string and replace that or something alike.